### ISSUE TITLE
🔦 Feature: Provide export access for instance users without backend access

### ISSUE THREAD

**Issue by vqqs3z9r** [nocodb/nocodb#8012]
### Please confirm if feature request does NOT exist already ?

- [X] I confirm there is no existing issue for this

### Describe the usecase for the feature

So that users of instances they aren't the owner of (e.g. the cloud service https://app.nocodb.com/) can backup and/or transfer their data (e.g. to a self-hosted nocodb instance).

### Suggested Solution

Provide mysqldump or pgdump functionality within the app itself, as this [appears](https://github.com/nocodb/nocodb/issues/384) to be the suggested backup solution for self-hosted instances, but of course that's not possible for instances hosted by someone else. To go with the backup, the download and/or interface should include the nocodb versions used on the instance, so that the user can ensure they import their backup into a compatible version.

Related issue: https://github.com/nocodb/nocodb/issues/8274

**Comment by eugenialoli** [comment]
Backing up a database to a real db file should be a feature by now. The maintainers have said many times that "this is not in the scope of the project", however, it is, under EU law. Exporting as CSV does not give you access to the metadata/relational info etc. It just dumps the rows, so it's an incomplete set of downloaded data under the law.

If all we wanted at the very end was a csv file, we would use Excel. We use NocoDB for its relational abilities. So these need to be exported too.

**Comment by springfielddatarecovery** [comment]
Looking into using nocodb for our CRM. This is a critical feature for us.

**Comment by dawidmachon** [comment]
Yes, be able to export from cloud to self-hosted is mandatory to be not lock in to vendor. And noco is solution to avoid lockin's...

**Comment by springfielddatarecovery** [comment]
We will pay anybody who writes this feature, submits it as a PR, and gets the code accepted into the codebase $25 in BTC. First one to get it done receives the payment.

Bounty expires Jun 2025

**Comment by springfielddatarecovery** [comment]
Bounty bumped to $100. Bounty will be paid out even if nocodb does not accept the PR. If they don't accept it because the code is of poor quality or doesn't work, the bounty will not be paid out, but if they refuse to accept the PR for some unreasonable reason (like they don't want this feature to exist), the bounty will still be paid out. Final discretion will rest with bounty sponsor (us).
